c2rust-analyze/src/rewrite/ty.rs

```diff
@@ -0,0 +1,394 @@
+    fn convert(self) -> ast::IntTy {
+        match self {
+            ty::IntTy::Isize => ast::IntTy::Isize,
+            ty::IntTy::I8 => ast::IntTy::I8,
+            ty::IntTy::I16 => ast::IntTy::I16,
+            ty::IntTy::I32 => ast::IntTy::I32,
+            ty::IntTy::I64 => ast::IntTy::I64,
+            ty::IntTy::I128 => ast::IntTy::I128,
+        }
+    }
+}
+
+impl Convert<ast::UintTy> for ty::UintTy {
+    fn convert(self) -> ast::UintTy {
+        match self {
+            ty::UintTy::Usize => ast::UintTy::Usize,
+            ty::UintTy::U8 => ast::UintTy::U8,
+            ty::UintTy::U16 => ast::UintTy::U16,
+            ty::UintTy::U32 => ast::UintTy::U32,
+            ty::UintTy::U64 => ast::UintTy::U64,
+            ty::UintTy::U128 => ast::UintTy::U128,
+        }
+    }
+}
+
+impl Convert<ast::FloatTy> for ty::FloatTy {
+    fn convert(self) -> ast::FloatTy {
+        match self {
+            ty::FloatTy::F32 => ast::FloatTy::F32,
+            ty::FloatTy::F64 => ast::FloatTy::F64,
+        }
+    }
+}
+
+impl Convert<hir::Mutability> for mir::Mutability {
+    fn convert(self) -> hir::Mutability {
+        match self {
+            mir::Mutability::Mut => hir::Mutability::Mut,
+            mir::Mutability::Not => hir::Mutability::Not,
+        }
+    }
+}
+
+fn mk_cell<'tcx>(tcx: TyCtxt<'tcx>, ty: ty::Ty<'tcx>) -> ty::Ty<'tcx> {
+    let core_crate = tcx
+        .crates(())
+        .iter()
+        .cloned()
+        .find(|&krate| tcx.crate_name(krate).as_str() == "core")
+        .expect("failed to find crate `core`");
+
+    let cell_mod_child = tcx
+        .module_children(core_crate.as_def_id())
+        .iter()
+        .find(|child| child.ident.as_str() == "cell")
+        .expect("failed to find module `core::cell`");
+    let cell_mod_did = match cell_mod_child.res {
+        Res::Def(DefKind::Mod, did) => did,
+        ref r => panic!("unexpected resolution {:?} for `core::cell`", r),
+    };
+
+    let cell_struct_child = tcx
+        .module_children(cell_mod_did)
+        .iter()
+        .find(|child| child.ident.as_str() == "Cell")
+        .expect("failed to find struct `core::cell::Cell`");
+    let cell_struct_did = match cell_struct_child.res {
+        Res::Def(DefKind::Struct, did) => did,
+        ref r => panic!("unexpected resolution {:?} for `core::cell::Cell`", r),
+    };
+
+    let cell_adt = tcx.adt_def(cell_struct_did);
+    let substs = tcx.mk_substs([GenericArg::from(ty)].into_iter());
+    tcx.mk_adt(cell_adt, substs)
+}
+
+/// Produce a `Ty` reflecting the rewrites indicated by the labels in `rw_lty`.
+fn mk_rewritten_ty<'tcx>(
+    lcx: LabeledTyCtxt<'tcx, RewriteLabel>,
+    rw_lty: RwLTy<'tcx>,
+) -> ty::Ty<'tcx> {
+    let tcx = *lcx;
+    lcx.rewrite_unlabeled(rw_lty, &mut |ty, args, label| {
+        let (own, qty) = match label.ty_desc {
+            Some(x) => x,
+            None => return ty,
+        };
+
+        assert_eq!(args.len(), 1);
```

Comment:
one argument because it's a pointer? if so can you add a comment explaining so?

Reply:
Added a comment.